File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<int> s;
        vector<int> waittime(temperatures.size());
        
        for(int i = 0; i < temperatures.size(); i++) {
            while(!s.empty() && temperatures[i] > temperatures[s.top()]) 
                waittime[s.top()] = i - s.top(), s.pop();
            s.push(i);
        }
        return waittime; 
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        s = []
        waittime = [0]*len(temperatures)
        
        for i, t in enumerate(temperatures):
            while s and t > temperatures[s[-1]]:
                waittime[s.pop()] = i - s[-1]
            s.append(i)
        return waittime
